src/Uno.UI/UI/Xaml/Controls/Border/Border.cs

```diff
@@ -296,5 +297,17 @@ protected virtual void OnBorderBrushChanged(Brush oldValue, Brush newValue)
 		#endregion
 
 		internal override bool CanHaveChildren() => true;
+
+		internal override bool IsViewHit() => IsViewHitImpl(this);
+
+		internal static bool IsViewHitImpl(FrameworkElement element)
+		{
+			_Debug.Assert(element is Panel
+				|| element is Border
+				|| element is ContentPresenter
+			);
+
+			return element.Background != null;
+		}
```

Comment:
@dr1rrb

Reply:
IsViewHit() is a close analogue of the `HitTestLocalInternal()` method from WinUI, probably doesn't make sense to remove it if our goal is to converge. I'll leave the future refactoring around it to @dr1rrb , though I will note that the `HitTestVisibilityProperty` machinery is a little problematic in terms of WinUI compatibility (eg it tightly couples the logical DP parent hierarchy to the visual parent hierarchy, which is not the case on WinUI).

It's in `Border` because it's in `Border` on WinUI.

Reply:
Didn't had chance to look in the WinUI code for that yet. But if there is a `HitTestLocalInternal` we will keep it. The question then would be to understand when it's invoked in WinUI. The advantage of the DP is that we are "pre-computing" the value, and we push it only once into native code (instead of invoking the `IsViewHit` for each pointer move with all the interop cost). But I agree it's definitely not the purpose of this PR.